mnefun/_mnefun.py

```diff
@@ -3650,3 +3686,54 @@ def extract_roi(stc, src, label=None, thresh=0.5):
     func_label = mne.Label(verts, hemi=hemi, subject=stc.subject)
     func_label = func_label.fill(src)
     return func_label, max_vert, max_vidx, max_tidx
+
+
+def do_mf_autobad(p, r, prebad_file):
+    t0 = time.time()
+    tmp_name = op.join(op.dirname(r), 'temp_%s_raw.fif' % t0)
+    tmp_raw = filter_chpi(read_raw_fif(r, allow_maxshield='yes',
+                                       preload=True),
+                          include_line=True)
+    mne.channels.fix_mag_coil_types(tmp_raw.info)
+    tmp_raw.save(tmp_name, fmt='short', overwrite=True)
+    remote_in = op.join(p.sws_ssh, p.sws_dir,
+                        op.basename(tmp_name))
+    print('     Copying...', end='')
+    cmd = ['scp', '-P', str(p.sws_port), tmp_name,
+           p.sws_ssh + ':' + p.sws_dir]
+    run_subprocess(cmd, stdout=subprocess.PIPE,
+                   stderr=subprocess.PIPE)
+    print('     Running maxfilter -autobad on', end='')
+    t1 = time.time()
+    tmp_name = op.join(op.dirname(r), 'temp_%s_raw.fif' % t1)
+    remote_out = op.join(p.sws_ssh, p.sws_dir,
+                         op.basename(tmp_name))
+    mf_log = op.join(p.sws_ssh, p.sws_dir,
+                     '%s_stdout.txt' % op.basename(remote_out[:-8]))
+    cmd = ['ssh', '-p', str(p.sws_port), p.sws_ssh,
+           '/neuro/bin/util/maxfilter -f ' + remote_in + ' -o ' +
+           remote_out + ' -autobad on -skip 0 30 -format short '
+                        '-badlimit %d -v' % p.mf_badlimit]
+    output = run_subprocess(cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.DEVNULL)[0]
```

Comment:
Can you make use of `run_sss_command` to avoid this boilerplate ssh-run-copy-delete stuff? You might have to modify `run_sss_command` a bit to get it to do what you want

Reply:
Initially tried getting `run_sss_comand` to do this but wasn't able to. I could give it another go but wasn't sure if you wanted the function modified and it was relatively less pain with `subprocess`.

Reply:
I'd rather get run_sss_command to have the right options to get what you need. That way the pushing, running, and deleting can be in one place